osu.Game.Rulesets.Mania/UI/DefaultColumn.cs

```diff
@@ -0,0 +1,104 @@
+    {
+        private readonly ColumnBackground background;
+        private readonly ColumnKeyArea keyArea;
+        private readonly ColumnHitObjectArea hitObjectArea;
+
+        // Pull up more stuff when needed.
+        protected override Container HitTargetContainer { get; }
+        protected override Drawable KeyArea => keyArea;
+        protected override Container ExplosionContainer { get; }
+        protected internal override Container TopLevelContainer { get; }
+
+        public DefaultColumn(int index)
+            : base(index)
+        {
+            background = new ColumnBackground { RelativeSizeAxes = Axes.Both };
+
+            InternalChildren = new[]
+            {
+                // For input purposes, the background is added at the highest depth, but is then proxied back below all other elements
+                background.CreateProxy(),
+                HitTargetContainer = new Container
+                {
+                    Name = "Hit target + hit objects",
+                    RelativeSizeAxes = Axes.Both,
+                    Children = new Drawable[]
+                    {
+                        hitObjectArea = new ColumnHitObjectArea(HitObjectContainer)
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                        },
+                        ExplosionContainer = new Container
+                        {
+                            Name = "Hit explosions",
+                            RelativeSizeAxes = Axes.Both,
+                        }
+                    }
+                },
+                keyArea = new ColumnKeyArea
+                {
+                    RelativeSizeAxes = Axes.X,
+                    Height = ManiaStage.HIT_TARGET_POSITION,
+                },
+                background,
+                TopLevelContainer = new Container { RelativeSizeAxes = Axes.Both }
+            };
+
+            TopLevelContainer.Add(ExplosionContainer.CreateProxy());
+        }
+
+        public override Axes RelativeSizeAxes => Axes.Y;
+
+        private Color4 accentColour;
+
+        public Color4 AccentColour
+        {
+            get => accentColour;
+            set
+            {
+                if (accentColour == value)
+                    return;
+
+                accentColour = value;
+
+                background.AccentColour = value;
+                keyArea.AccentColour = value;
+                hitObjectArea.AccentColour = value;
+            }
+        }
+
+        /// <summary>
+        /// Adds a DrawableHitObject to this Playfield.
+        /// </summary>
+        /// <param name="hitObject">The DrawableHitObject to add.</param>
+        public override void Add(DrawableHitObject hitObject)
+        {
+            hitObject.AccentColour.Value = AccentColour;
+            base.Add(hitObject);
+        }
+
+        public override void AddHitExplosion(DrawableHitObject judgedObject)
```

Comment:
This should only be in the base class. The hit explosion should be a skinnable drawable itself.